lib/astarte_core/cql_utils.ex

```diff
@@ -167,4 +167,35 @@ defmodule Astarte.Core.CQLUtils do
 
     eid
   end
+
+  @spec realm_name_to_keyspace_name(nonempty_binary(), nonempty_binary()) :: nonempty_binary()
+  def realm_name_to_keyspace_name(realm_name, astarte_instance_id \\ "default")
+      when is_binary(realm_name) do
+    case :persistent_term.get(realm_name, nil) do
+      nil ->
+        if String.length(astarte_instance_id <> realm_name) < 48 do
+          keyspace_name =
+            (astarte_instance_id <> realm_name)
+            |> String.replace("-", "")
+            |> String.replace("_", "")
+            |> String.downcase()
+
+          :persistent_term.put(realm_name, keyspace_name)
+          keyspace_name
+        else
+          keyspace_name =
+            Base.url_encode64(astarte_instance_id <> realm_name, padding: false)
+            |> String.replace("-", "")
+            |> String.replace("_", "")
+            |> String.slice(0..47)
+            |> String.downcase()
+
+          :persistent_term.put(realm_name, keyspace_name)
+          keyspace_name
```

Comment:
I'd isolate the pure function from the side effects, i.e. `realm_name_to_keyspace_name/2` just transforms the realm name in a keyspace name, and then `persistent_realm_name_to_keyspace_name/2` gets it from persistent term (calculating it using the previous function if not found).
This also allows you to unit test the logic of the pure function without worrying about persistent terms.

Reply:
(the names are a suggestion, I'm open to other proposals too)

Reply:
@rbino the idea behind the current impl is that it's faster to retrieve an already encoded realm name from persistent_terms rather than re-encode each time it is accessed. 
Probabily it's better to just rename the function in persistent_realm_name_to_keyspace_name and create a private one for the encoding? since it's useless to encode without check persistent_term first outside of this module